### ISSUE TITLE
Jellyfin plugin is not working

### ISSUE THREAD

**Issue by florian-asche** [music-assistant/hass-music-assistant#2383]
### What version of Music Assistant has the issue?

2.1.0b1

### What version of the Home Assistant Integration have you got installed?

2024.5.1

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

MusicAssistant does not show any media from Jellyfin.

### How to reproduce

I am using Jellyfin 10.9.2. I connected MusicAssistant with its own user via http to Jellyfin. The user has admin permission and has the permission to download content. I only allowed him to access music content.

Jellyfin and musicassistant is running as a docker instance. 

### Music Providers

Jellyfin

### Player Providers

Snapcast

### Full log output

[_music-assistant-server_logs.txt](https://github.com/music-assistant/hass-music-assistant/files/15475825/_music-assistant-server_logs.txt)

### What version of Home Assistant Core are your running

2024.5.4

### What type of installation are you running?

Home Assistant Container

### On what type of hardware are you running?

Linux

**Comment by florian-asche** [comment]
As already suggested in other threads, i shared the media directly to musicassistant, and the media is there and able to play.

**Comment by florian-asche** [comment]
I set the log level to debug inside the music provider setup:
Logs from MusicAssistant:
```
2024-05-28 20:55:10.441 INFO (ThreadPoolExecutor-0_4) [JELLYFIN.jellyfin_apiclient_python.api] Sending get request to system/info/public
2024-05-28 20:55:10.445 INFO (ThreadPoolExecutor-0_4) [JELLYFIN.jellyfin_apiclient_python.connection_manager] connect_to_address http://192.168.1.5:8596 succeeded
2024-05-28 20:55:10.445 INFO (ThreadPoolExecutor-0_4) [JELLYFIN.jellyfin_apiclient_python.connection_manager] begin connect_to_server
2024-05-28 20:55:10.445 INFO (ThreadPoolExecutor-0_4) [JELLYFIN.jellyfin_apiclient_python.api] Sending get request to system/info/public
2024-05-28 20:55:10.448 INFO (ThreadPoolExecutor-0_4) [JELLYFIN.jellyfin_apiclient_python.connection_manager] calling onSuccessfulConnection with server None
2024-05-28 20:55:10.448 INFO (ThreadPoolExecutor-0_4) [JELLYFIN.jellyfin_apiclient_python.credentials] {}
2024-05-28 20:55:10.448 INFO (ThreadPoolExecutor-0_4) [JELLYFIN.jellyfin_apiclient_python.api] Trying to login to http://192.168.1.5:8596/Users/AuthenticateByName as musicassistant
2024-05-28 20:55:10.448 INFO (ThreadPoolExecutor-0_4) [JELLYFIN.jellyfin_apiclient_python.api] Sending post request to Users/AuthenticateByName
2024-05-28 20:55:10.593 INFO (ThreadPoolExecutor-0_4) [JELLYFIN.jellyfin_apiclient_python.connection_manager] Succesfully logged in as musicassistant
2024-05-28 20:55:10.593 INFO (MainThread) [music_assistant] Loaded music provider Jellyfin Media Server Library
2024-05-28 20:55:12.041 INFO (MainThread) [music_assistant.music] Sync task for Jellyfin Media Server Library completed
```
Logs from Jellyfin:
```
[2024-05-28 22:42:07.399 +02:00] [INF] [109] Jellyfin.Api.Auth.CustomAuthenticationHandler: AuthenticationScheme: "CustomAuthentication" was challenged.
[2024-05-28 22:42:38.725 +02:00] [INF] [83] Jellyfin.Server.Implementations.Users.UserManager: Authentication request for "musicassistant" has succeeded.
[2024-05-28 22:42:38.725 +02:00] [INF] [83] Emby.Server.Implementations.Session.SessionManager: Current/Max sessions for user "musicassistant": 8/0
[2024-05-28 22:42:38.726 +02:00] [INF] [83] Emby.Server.Implementations.Session.SessionManager: Creating new access token for user 5863345a-8bd5-43c3-8471-2a2ee8dbe44a
[2024-05-28 22:42:38.787 +02:00] [INF] [83] Emby.Server.Implementations.HttpServer.WebSocketManager: WS "192.168.1.20" request
[2024-05-28 22:43:27.060 +02:00] [INF] [103] Jellyfin.Server.Implementations.Users.UserManager: Authentication request for "musicassistant" has succeeded.
[2024-05-28 22:43:27.060 +02:00] [INF] [103] Emby.Server.Implementations.Session.SessionManager: Current/Max sessions for user "musicassistant": 9/0
[2024-05-28 22:43:27.061 +02:00] [INF] [103] Emby.Server.Implementations.Session.SessionManager: Creating new access token for user 5863345a-8bd5-43c3-8471-2a2ee8dbe44a
[2024-05-28 22:55:10.586 +02:00] [INF] [21] Jellyfin.Server.Implementations.Users.UserManager: Authentication request for "musicassistant" has succeeded.
[2024-05-28 22:55:10.586 +02:00] [INF] [21] Emby.Server.Implementations.Session.SessionManager: Current/Max sessions for user "musicassistant": 10/0
[2024-05-28 22:55:10.587 +02:00] [INF] [21] Emby.Server.Implementations.Session.SessionManager: Creating new access token for user 5863345a-8bd5-43c3-8471-2a2ee8dbe44a
```

**Comment by OzGav** [comment]
@lokiberra

**Comment by florian-asche** [comment]
Looking at my metadataprovider in jellyfin, i only used MusicBrainz, and no TheAudioDB. I removed all my media from jellyfin and added just one album for testing. I selected both metadata providers.

**Comment by florian-asche** [comment]
I also used Picard with ID3v2.4 tags.

Sadly the media is still not in musicAssistant :/

**Comment by florian-asche** [comment]
I update jellyfin to 10.9.3. Same error :(

**Comment by lokiberra** [comment]
Thanks for the log. Based on the last two lines it connects just fine to your instance and is supposed to start syncing items. 

The "Loaded music provider Jellyfin Media Server Library" comes from load_provider in server.py. It should then try to start syncing artists as the first media_type. If it does it should give a message of "Start sync of Artists items." as part of the sync_library in music_provider.py. 

If it is successful in running all the tasks for each media_type it will return "Sync task for Jellyfin Media Server Library completed" as part of the _start_provider_sync in music.py. 

Long story short, based on the lack of messages in the log it looks like something is happening in the main MA server code base apart from the Jellyfin provider itself that prevents the sync tasks from even starting.

@OzGav  I can't tell what might be going wrong between load_provider and the sync_library task from starting and actually using the Jellyfin provider

**Comment by OzGav** [comment]
@lokiberra ok sounds like we will need to wait for this to get to the top of the pile and Marcel will comment.

**Comment by OzGav** [comment]
This is being looked at

**Comment by marcelveldt** [comment]
The Jellyfin provider has had a huge bump of updates last couple of weeks and months thanks to @Jc2k.

@florian-asche can you please verify if your issues still exist with the most recent 2.2.0 update of MA Server ?

**Comment by florian-asche** [comment]
> The Jellyfin provider has had a huge bump of updates last couple of weeks and months thanks to @Jc2k.
> 
> @florian-asche can you please verify if your issues still exist with the most recent 2.2.0 (or 2.3 beta) update of MA Server ?

I am currently running 2.2.2 and enabled the jellyfin addon. Good news is, data can be found. I let it scan my library now. Will report if the time that it needs is ok and so on.

**Comment by florian-asche** [comment]
> > The Jellyfin provider has had a huge bump of updates last couple of weeks and months thanks to @Jc2k.
> > @florian-asche can you please verify if your issues still exist with the most recent 2.2.0 (or 2.3 beta) update of MA Server ?
> 
> I am currently running 2.2.2 and enabled the jellyfin addon. Good news is, data can be found. I let it scan my library now. Will report if the time that it needs is ok and so on.

it run trough. Looks great. Thanks for the good work!
